sources/HUBViewController.m

```diff
@@ -708,19 +706,16 @@ - (UICollectionViewCell *)collectionView:(UICollectionView *)collectionView cell
     
     HUBComponentCollectionViewCell * const cell = [collectionView dequeueReusableCellWithReuseIdentifier:cellReuseIdentifier
                                                                                             forIndexPath:indexPath];
-    
-    if (cell.component == nil) {
-        HUBComponentWrapper * const componentWrapper = [self.componentReusePool componentWrapperForModel:componentModel
-                                                                                                delegate:self
-                                                                                                  parent:nil];
-        
-        self.componentWrappersByCellIdentifier[cell.identifier] = componentWrapper;
-        cell.component = componentWrapper;
-        [componentWrapper viewDidMoveToSuperview:cell];
-        [self didAddComponentWrapper:componentWrapper];
-    }
-    
-    HUBComponentWrapper * const componentWrapper = [self componentWrapperFromCell:cell];
+
+    HUBComponentWrapper * const componentWrapper = [self.componentReusePool componentWrapperForModel:componentModel
+                                                                                            delegate:self
+                                                                                              parent:nil];
+
+    self.componentWrappersByCellIdentifier[cell.identifier] = componentWrapper;
```

Comment:
Are we ever clearing this up? That is, before we assign a new wrapper to the cell, we should do:
```
self.componentWrappersByCellIdentifier[cell.identifier] = nil;
```